### DISCUSSION TITLE
An option to add variable to the watch window

### DISCUSSION THREAD

**Discussion by Momoshikin** [zero-plusplus/vscode-autohotkey-debug#153] [Ideas]
Is possible somehow to add a variable to the watch window like on visual studio:
![XjrwgOvZSY](https://user-images.githubusercontent.com/92819425/138161851-ac40eb97-663e-4498-8136-fcaadc9229fc.gif)

I couldn't find this option here:
![2021-10-20_16-51-06](https://user-images.githubusercontent.com/92819425/138163116-c02e9533-851f-43fd-8987-ec342ed37cf2.gif)

So I think there's no support for it yet, would like to request this option if possible.

And if I could suggest something more, would be a better way to see the content of a variable, for example:

![image](https://user-images.githubusercontent.com/92819425/138162970-c2cef5a5-38a2-4ade-a6d2-93e62b99b781.png)

When I need to copy the content of a variable directly from the debugger panel it sucks.

Good job with the extension, I started using it now, and I'm loving 👍 💯

**Comment by zero-plusplus** [comment]
**I use DeepL translation.**

Thank you for using my extension.

The first one will be enabled in the next version. Also, you can register variables to `watch expressions` from the variables view.

The second one is under consideration but not yet implemented. 
Since VSCode does not have the feature to create a window, the destination for the display will be the editor. If you're okay with that, it can be implemented.

**Reply by Momoshikin** [reply]
Cool! When this new version will be launched?

**Reply by zero-plusplus** [reply]
The next version is scheduled for sometime in November.
As this is a relatively large update, a lot of work needs to be done to validate and test the new features. As a result, it may be later than planned.

The work is managed as [here](https://github.com/zero-plusplus/vscode-autohotkey-debug/issues/144).

**Reply by zero-plusplus** [reply]
@Momoshikin
`Add to Watch` has been implemented in `1.10.0`.

This context menu is available from within the editor or Variables view.
